src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs

```diff
@@ -302,22 +303,19 @@ public async Task<IReadOnlyCollection<ChannelSummaryResponse>> GetAllowedChannel
             var channels = await UnitOfWork.ChannelRepository.GetAllowedChannelsWithMessagesAndCreatorAsync(currentUser.Id);
             foreach (var channel in channels)
             {
-                var channelMember = await UnitOfWork.ChannelMemberRepository.GetChannelMemberAsync(currentUser.Id, channel.Id);
+                channel.Messages = channel.Messages?.Where(x => x.AccessibilityStatus == AccessibilityStatus.Present).ToList();
                 if (channel.Type == ChannelType.Direct)
                 {
-                    //TODO: Extend existing query to make one single request to get all data for both direct and group channel
                     var members = await UnitOfWork.ChannelMemberRepository.GetChannelMembersWithMemberDetailsAsync(channel.Id);
-                    var directChannelMember = members.SingleOrDefault(x => x.MemberId != currentUser.Id);
-                    if (directChannelMember == null)
-                    {
-                        throw new NetKitChatNotFoundException($"Direct channel member is not found.");
-                    }
+                    var existingUserChannelMember = members.First(x => x.MemberId == currentUser.Id);
```

Comment:
Let's use FirstOrDefault() action. Also we need to check entity on null.
Let's check directChannelMember variable also.

Reply:
In this case we don't expect exceptional behavior. Chat between two members should contain both members including existing user otherwise it's better to throw exception.